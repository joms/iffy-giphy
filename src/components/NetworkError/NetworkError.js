import React from 'react';
import PropTypes from 'prop-types';

const NetworkError = ({ error }) => (
    <h1>
        Error {error.status}: {error.message}
    </h1>
);

NetworkError.propTypes = {
    error: PropTypes.shape({
        message: PropTypes.string,
        ok: PropTypes.bool,
        status: PropTypes.number,
        statusText: PropTypes.string,
    }).isRequired,
};

export default NetworkError;
