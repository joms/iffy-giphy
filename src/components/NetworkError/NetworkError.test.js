import React from 'react';
import { shallow } from 'enzyme';
import NetworkError from '.';

describe('<NetworkError>', () => {
    const error = { status: 401, ok: false, statusText: '', message: 'No API key found in request' };
    const wrapper = shallow(<NetworkError error={error} />);
    it('matches snapshot', () => {
        expect(wrapper).toMatchSnapshot();
    });

    it('renders the message', () => {
        expect(wrapper.text().includes(error.message)).toBeTruthy();
        expect(wrapper.text().includes(error.status)).toBeTruthy();
    });
});
