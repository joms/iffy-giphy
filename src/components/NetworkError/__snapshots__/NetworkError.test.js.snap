// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NetworkError> matches snapshot 1`] = `
<h1>
  Error 
  401
  : 
  No API key found in request
</h1>
`;
